R/predict.ergm.R

```diff
@@ -0,0 +1,75 @@
+#' ERGM-based conditional tie probabilities
+#'
+#' This will calculate **conditional** tie probabilities for all dyads in `net`
+#' based on the ERGM model `fit`.
+#'
+#' @param fit fitted ERGM model object
+#' @param net network object
+#'
+#' @return 
+#' N x N matrix of conditional tie probabilities, where N is the number of nodes in `net`.
+#' 
+#' @method predict ergm
+#'
+#' @export
+
+predict.ergm <- function(fit, net) {
+  # Argument checking
+  stopifnot(inherits(fit, "ergm"))
+  stopifnot(inherits(net, "network"))
+  if( is.directed(fit$network) != is.directed(net) )
+    stop(
+      "the model was fit to ",
+      if(is.directed(fit$network)) "directed" else "undirected",
+      " network while `net` is ",
+      if(is.directed(net)) "directed" else "undirected"
+    )
+  # Fix/enformulate curved ERGMs if necessary
+  if(is.curved(fit)) {
+    fixed <- fix.curved(fit)
+    coeff <- fixed$theta
+    form <- fixed$formula
+  } else {
+    coeff <- coef(fit)
+    form <- fit$formula
+  }
+  n <-  network.size(net)
+  # Substitute network object in formula's environment
+  e <- rlang::env(net = net)
```

Comment:
I think this would clobber formula's original environment, which could break terms whose arguments were variables. `statnet.common::nonsimp_update.formula()` might do the trick. Or, does `rlang` have some utilities?

Reply:
Yeah.. I was wondering about that as I noticed that `ergm` extends standard formulas in some ways. However, this only substitutes a different network object (called `net` on LHS anyway) so that I can call `summary(form)` on different "toggled" networks. It does not touch RHS nor any other objects in the formula-associated environment (if there are any).

Reply:
It's not so much that `ergm` extends formulas as that it uses it in specific ways. In this case, though, if there are terms in the formula that contain variables present in the formula environment, it may cause them to be missed.